coreos-iso-embed-ignition

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python3
+
+import argparse
+import gzip
+import os
+import shutil
+import struct
+import subprocess
+import sys
+import tempfile
+
+FILENAME = 'config.ign'
+
+class EmbedException(Exception):
+    pass
+
+
+class CopiedFileHolder:
+    '''
+    Holder for a file handle which can optionally be copied to another one,
+    then modified, then finalized.  If copied but not finalized, the copy
+    will be deleted when GCed.
+    '''
+    def __init__(self, fh):
+        self.fh = fh
+        self._copied = False
+        self._complete = False
+
+    def __del__(self):
+        if self._copied and not self._complete:
+            os.remove(self.fh.name)
+
+    def copy_to(self, target):
+        assert not self._copied
+        assert not self._complete
+        src = self.fh
+        self.fh = target
+        self._copied = True
+        shutil.copyfileobj(src, target)
+
+    def finish(self):
+        self._complete = True
+
+
+def make_cpio(ignition_fh):
+    '''
+    Make a gzipped CPIO archive containing the specified Ignition config.
+    '''
+    with tempfile.TemporaryDirectory(prefix='coreos-iso-') as tempdir:
+        config_path = os.path.join(tempdir, FILENAME)
+        with open(config_path, 'w') as out_fh:
+            shutil.copyfileobj(ignition_fh, out_fh)
+        proc = subprocess.Popen(['cpio', '-o', '-D', tempdir, '-H', 'newc',
+                '--quiet'], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        archive, _ = proc.communicate(FILENAME.encode())
+        if proc.returncode != 0:
+            raise EmbedException(f'cpio exited with status {proc.returncode}.')
+    return gzip.compress(archive)
+
+
+def extract_cpio(buf):
+    '''
+    Extract a gzipped CPIO archive and return the contents of the Ignition
+    config.
+    '''
+    cpio = gzip.decompress(buf)
+    with tempfile.TemporaryDirectory(prefix='coreos-iso-') as tempdir:
+        proc = subprocess.Popen(['cpio', '-i', '-D', tempdir, '--quiet',
+                FILENAME], stdin=subprocess.PIPE)
```

Comment:
Optional: could probably use `--to-stdout` here and just `TemporaryFile` directly. If the OS supports `O_TMPFILE`, it avoids creating a user visible inode entirely.